ministryofjustice/correspondence_tool_staff
# Adds CRN number to Offender SAR cases

## Description
Adds CRN number to Offender SAR cases. This is so users can store the relevant identifying information about a data subject, and this data can be searched on later

### Questions to guide a reviewer:
- I have added a test for the details page explicitly. It seems that there are few explicit tests for the `offender_sar/new` page. Tests for create use page_objects which build using the factories. I have updated the factory to include the new field and things seem to pass. I would like your opinion on whether this is the right approach here and whether I can do more on testing the input side explicitly.
- I noticed that there is a `searchable_fields_and_ranks` method - which I am assuming sets out a priority map for data points that will be returned on a search by a user. I haven't added the CRN number to this yet, as I am still not fully aware of the context, but would this be something that I should do? (either here or in a separate PR).


## Self-review checklist


### Screenshots
**Before:**

<img width="609" alt="Screenshot 2020-06-30 at 17 26 55" src="https://user-images.githubusercontent.com/13377553/86151695-fd24f400-baf6-11ea-89cb-0fd3d98343a9.png">

**After:**

<img width="596" alt="Screenshot 2020-06-30 at 17 28 12" src="https://user-images.githubusercontent.com/13377553/86151799-1e85e000-baf7-11ea-95f3-c6a92ba5033c.png">

*case details:*
<img width="991" alt="Screenshot 2020-06-30 at 17 37 20" src="https://user-images.githubusercontent.com/13377553/86152695-7bce6100-baf8-11ea-8cfb-ee90552a8d2d.png">

### Related JIRA tickets
[CT-2892](https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2892)

### Deployment
- no new migrations as this adds to the JSONB field in the Case table.

### Manual testing instructions
1. Login as user with Applications privileges i.e. Brian Rix
2. Click create a case
3. Add relevant details (including CRN)
4. Click through the Wizard
5. You should see the CRN number appear on the case details page at the end of the flow.
